diffrax/_term.py

```diff
@@ -776,3 +785,135 @@ def _to_vjp(_y, _diff_args, _diff_term):
         dy, vjp = jax.vjp(_to_vjp, y, diff_args, diff_term)
         da_y, da_diff_args, da_diff_term = vjp((-(a_y**ω)).ω)
         return dy, da_y, da_diff_args, da_diff_term
+
+
+LangevinX = PyTree[Shaped[Array, "?*langevin"], "LangevinX"]
+LangevinTuple = tuple[LangevinX, LangevinX]
+_LangevinBM = TypeVar("_LangevinBM", bound=Union[LangevinX, AbstractBrownianIncrement])
+
+
+class _LangevinDiffusionTerm(AbstractTerm[LangevinX, _LangevinBM]):
+    r"""Represents the diffusion term in the Langevin SDE:
+
+    $d \mathbf{x}_t = \mathbf{v}_t \, dt$
+
+    $d \mathbf{v}_t = - \gamma \, \mathbf{v}_t \, dt - u \,
+    \nabla \! f( \mathbf{x}_t ) \, dt + \sqrt{2 \gamma u} \, d W_t.$
+    """
+
+    gamma: LangevinX
+    u: LangevinX
+    control: AbstractBrownianPath
+
+    def __init__(self, gamma, u, control: AbstractBrownianPath):
+        self.gamma = gamma
+        self.u = u
+        self.control = control
+
+    def vf(self, t: RealScalarLike, y: LangevinTuple, args: Args) -> LangevinX:
+        x, v = y
+
+        def _fun(_gamma, _u, _v):
+            return jnp.sqrt(2 * _gamma * _u) * jnp.ones(_v.shape, _v.dtype)
+
+        d_v = jtu.tree_map(_fun, self.gamma, self.u, v)
+        return d_v
+
+    def contr(self, t0: RealScalarLike, t1: RealScalarLike, **kwargs) -> _LangevinBM:
+        return self.control.evaluate(t0, t1, **kwargs)
+
+    def prod(self, vf: LangevinX, control: LangevinX) -> LangevinTuple:
+        if isinstance(control, AbstractBrownianIncrement):
+            control = control.W
+
+        dv = vf
+        dw = control
+        v_out = jtu.tree_map(operator.mul, dv, dw)
+        x_out = jtu.tree_map(jnp.zeros_like, v_out)
+        return x_out, v_out
+
+
+_LangevinArgs = tuple[LangevinX, LangevinX, Callable[[LangevinX], LangevinX]]
+
+
+def _langevin_drift(t, y: LangevinTuple, args: _LangevinArgs) -> LangevinTuple:
+    gamma, u, grad_f = args
+    x, v = y
+    f_x = grad_f(x)
+    d_x = v
+    d_v = jtu.tree_map(
+        lambda _gamma, _u, _v, _f_x: -_gamma * _v - _u * _f_x,
+        gamma,
+        u,
+        v,
+        f_x,
+    )
+    d_y = (d_x, d_v)
+    return d_y
+
+
+def _broadcast_pytree(source, target_tree_shape):
+    # Requires that source is a prefix tree of target_tree_shape
+    def _inner_broadcast(_src_arr, _inner_tree_shape):
+        _arr = jnp.asarray(_src_arr)
+        return jtu.tree_map(
+            lambda _leaf: jnp.broadcast_to(_arr, _leaf.shape), _inner_tree_shape
+        )
+
+    return jtu.tree_map(_inner_broadcast, source, target_tree_shape)
+
+
+class LangevinTerm(AbstractTerm):
```

Comment:
I introduced `LangevinTerm` for the reasons below, but I am happy to change this if you think there's a better way to handle this.
The requirements are:
(1) this term can be passed into any of the regular SRKs, which only accept a
MultiTerm[ODETerm, AbstractTerm]
(2) this term needs to carry an `args` attribute. You said that
since ALIGN, SORT and ShOULD, use the `args` in a very special way, we
cannot just pass them as usual (i.e. in `vf(t, y, args)`), but that the args
should be part of the term structure itself.

Frankly, I could just use a tuple (args, MultiTerm[ODETerm, AbstractTerm])
and pass that around, but the issue is that if the user mishandles
the tuple (e.g. changes the args but not the term), then the solution from
ALIGN (which depends on the args) will be different than the one from SRK,
which depends only on the MultiTerm (which has the args baked into the VF).

In a way `LangevinTerm` is just a wrapper for a `MultiTerm`.

Reply:
I think having this as a separate term seems totally reasonable to me.

Reply:
Although, looking at what you have here: this is basically a custom constructor for a `MultiTerm`, right?

In that case why not just make this a function, rather than a class, which returns a `MultiTerm` with the appropriate inputs?

Reply:
No, that's my whole point: it's not just a custom `MultiTerm` constructor. It is a container that holds both the arguments `(gamma, u, grad_f)` and the `MultiTerm`. The latter can be computed from the former, but not the other way round (at least not in a simple way). And `AbtractLangevinSRK` solvers need just the arguments, whereas all other solvers need just the `MultiTerm`. Hence both of these objects need to be carried around and then when `diffeqsolve` is called, depending on the solver, it either passes the `MultiTerm` or the arguments. I am very strongly convinced that the correct behaviour here cannot be achieved without both modifying `diffeqsolve` and introducing a new term. Maybe we should have a call to discuss this.